mapper.go

```diff
@@ -256,6 +295,93 @@ func mapRowSerial(
 	return row.Scan(destVals...)
 }
 
+// mapOuterJoinRowSerial maps rows to **Model. NULL value in ColumnSplitter column sets *Model = nil.
+// It can map inner join rows as well but using mapRowSerial is faster because less reflection logics.
+func mapOuterJoinRowSerial(
+	row *sql.Rows,
+	destList []*reflect.Value,
+	headColProvider ColumnSplitter,
+) error {
+	// **Model
+	var destFields []map[string]int
+	destTypes := map[int]reflect.Type{}
+	for destIndex, dest := range destList {
+		fields, err := aggregateFields(dest)
+		if err != nil {
+			return err
+		}
+		destFields = append(destFields, fields)
+		destTypes[destIndex] = dest.Type().Elem() // Model
+	}
+	cols, err := row.ColumnTypes()
+	if err != nil {
+		return err
+	}
+	destVals := make([]interface{}, len(cols))
+	colIndex := 0
+	for destIndex := range destList {
+		fields := destFields[destIndex]
+		headCol := cols[colIndex]
+		expectedHeadCol := headColProvider(destIndex)
+		if headCol.Name() != expectedHeadCol {
+			return fmt.Errorf(
+				"head col mismatch: expected=%s, actual=%s",
+				expectedHeadCol, headCol.Name(),
+			)
+		}
+		start := colIndex
+		ns := &noopScanner{}
+		model := reflect.New(destTypes[destIndex]).Elem() // Model
+		for ; colIndex < len(cols); colIndex++ {
+			col := cols[colIndex]
+			if colIndex > start && destIndex < len(destList)-1 {
+				// Reach next column's head
+				if col.Name() == headColProvider(destIndex+1) {
+					break
+				}
+			}
+			if fIndex, ok := fields[col.Name()]; ok {
+				f := model.Field(fIndex)
+				value := reflect.New(f.Addr().Type()) // **(Model.Field)
+				destVals[colIndex] = value.Interface()
+			} else {
+				destVals[colIndex] = ns
+			}
+		}
+	}
+	if err := row.Scan(destVals...); err != nil {
+		return err
+	}
+
+	colIndex = 0
+	for destIndex, dest := range destList {
+		if reflect.ValueOf(destVals[colIndex]).Elem().IsNil() { // *(Model.Field) == nil
+			colIndex += len(destList)
+			dest.Set(reflect.Zero(dest.Type()))
+			continue
+		}
+		model := reflect.New(destTypes[destIndex]) // *Model
```

Comment:
Does the original pointed value is always ignored? Isn't it possible to use original struct's fields if they exist?

```go
org := Value{}
orgPtr := &org
m.Map(rows, &orgPtr)
// It's desirable that &org == orgPtr even after mapping?
```

Reply:
It's possible but it means

* When `var orgPtr *Value` is used, always use `dest.Set(model)` 
* When `orgPtr := &Value{}` is used,
    * If data exists in the row, set each orgPtr.fields by  `f.Set`
    * If not exists, use `orgPtr.Set(nil value)`

I thought using `dest.Set(model)` in both cases was easier to read and maintain code.
What do you think? Do you think `&org == orgPtr even after mapping?` is more important in this case?

Reply:
I felt it's opaque behavior that pointer value is swapped by reflection. I agree that in the 99% of cases, there are few reason to use pointer to existing value. So how about accepting only pointer of pointer to nil?

Reply:
updated.